.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,30 @@
+name: Integration Tests
+
+on:
+    - pull_request
+    - push
+
+jobs:
+  dispatch:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get the branch name
+        id: get_branch
+        run: echo ::set-output name=branch::$(echo ${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}})
+
+      - name: Get the originating repository
+        id: get_repo
+        run: |
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            echo ::set-output name=repo::${{ github.event.pull_request.head.repo.full_name }}
+          else
+            echo ::set-output name=repo::${{ github.repository }}
+          fi
+
+      - name: Repository Dispatch
+        uses: peter-evans/repository-dispatch@v1
```

Comment:
This is very risky. First any push that `peter-evans` & collaborators does do the `v1` tag will automagically run in our env and secondly we give this action access to the PAT. If we are going to rely on any third-party actions we need to commit hash pin them (which is it's own sort of maintenance pain).

Reply:
Cool, so now that it is pinned, do we have a plan for how we want to update it? I went and read the source, it is pretty straight forward so that is good, but do we need to add dependabot or something similar to update these hashes?

Reply:
Great point! I added the pin in b6d2038 (also upgraded to 3.0.0). I used a fine token with only access to `ulisesgascon/express-examples` (`expressjs/examples` in the near future) with content write and read level permissions.

Managing upgrades will be easier than expected as #5435 will generate PRs that supports the pinned version format.

Reply:
hmmm, this seems suspicious though, won;t this log the token? https://github.com/peter-evans/repository-dispatch/blob/main/src/main.ts#L23

Reply:
@wesleytodd yes, though it'd be redacted via GHA (most likely). However, that's a non-starter IMO. 

I believe my team recently implemented this without using this dispatch method, complete with "waiting" and access to the results. Let me sync with them and see if there's a solution to be distilled there (or if it was trivially easy)

Reply:
@dwelch2344 is right and I will love to know more about the sync 👍 .

@wesleytodd, I assume that you can't access to the logs in https://github.com/UlisesGascon/express/actions/runs/7950651381/job/21703181190 (last run), but in this screenshot we can see the token redacted.
![Screenshot 2024-02-20 at 19 59 31](https://github.com/expressjs/express/assets/5110813/7b8f4cf1-603d-4c7f-b3e6-61aee3cff27b)

> Warning: If a secret was used in the job, GitHub automatically redacts secrets printed to the log. You should avoid printing secrets to the log intentionally. [Documentation](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions)

Reply:
Ok cool, I think it is also reasonable to PR to them to just remove it explicitly from the debug log. I didn't go look at the runs I was just poking around the code as I would do with any added dependency and that line stood out to me. It is good that GHA redacts that, so that resolves that concern for me.

For the dependabot thing, I raised my concern there and we can have that conversation independent of this landing. Consider my concerned addressed but would also be interested in the details from @dwelch2344 so we can leave this thread open for now.

Reply:
I removed the external dependency and used curl to manage the lifecycle (https://github.com/expressjs/express/pull/5492/commits/41cecb0c57c8b4372d99d775fbae3082729cd229). I will add screenshots in a moment